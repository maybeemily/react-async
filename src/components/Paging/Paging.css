.Paging {
    margin: 1vw;
  }
