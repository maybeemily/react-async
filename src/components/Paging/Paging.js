import React from 'react';
import PropTypes from 'prop-types';
import styles from './Paging.css';

function Paging({ currentPage, totalPages, nextPage, children }) {
  return (
    <>
    {currentPage > 1 && 
      <button className={styles.Paging} onClick={() => nextPage(currentPage - 1)}>PreviousPage</button>}
      {`${currentPage} / ${totalPages}`}
      {currentPage < totalPages && 
      <button className={styles.Paging} onClick={() => nextPage(currentPage + 1)}>Next Page</button>}
      {children}
      </>
  );
}

Paging.propTypes = {
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  nextPage: PropTypes.func.isRequired,
  children: PropTypes.node
};

export default Paging;
