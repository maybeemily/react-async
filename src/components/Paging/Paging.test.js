import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

//renders the paging - currentPage, totalPages, nextPage, children

describe('Paging component', () => {
  it('renders the paging', () => {
    const wrapper = shallow(<Paging currentPage={1} totalPages={10} nextPage={jest.fn()}>
      <h1>children!</h1>
    </Paging>);
    expect(wrapper).toMatchSnapshot();
  });
});
