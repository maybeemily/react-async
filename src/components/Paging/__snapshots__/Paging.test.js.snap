// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders the paging 1`] = `
<Fragment>
  1 / 10
  <button
    onClick={[Function]}
  >
    Next Page
  </button>
  <h1>
    children!
  </h1>
</Fragment>
`;
